pkg/apis/gcp/validation/controlplane.go

```diff
@@ -35,6 +36,11 @@ func ValidateControlPlaneConfig(controlPlaneConfig *apisgcp.ControlPlaneConfig,
 	if !workerZones.Has(controlPlaneConfig.Zone) {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("zone"), controlPlaneConfig.Zone, "must be part of at least one worker zone"))
 	}
+
+	if controlPlaneConfig.CloudControllerManager != nil {
```

Comment:
Do we need to do this also for the update scenario below? Or is this method also called when a `shoot` is updated?

Reply:
This method is also called when a `shoot` is updated (via `validateConext`).